glue/core/data.py

```diff
@@ -1825,12 +1828,20 @@ def compute_statistic(self, statistic, cid, subset_state=None, axis=None,
             # shape of the full results had subarray_slices not been set,
             # then insert the result into it. If axis is None, then we don't
             # need to do anything, and this is covered by the first clause
-            # of the if statement above.
+            # of the if statement above. Likewise if a view was specified,
+            # only the result within the view is returned.
             if not isinstance(axis, tuple):
                 axis = (axis,)
-            full_shape = [self.shape[idim] for idim in range(self.ndim) if idim not in axis]
-            full_result = np.zeros(full_shape) * np.nan
             result_slices = tuple([subarray_slices[idim] for idim in range(self.ndim) if idim not in axis])
+
+            if chunk_view is None:
+                full_shape = [self.shape[idim] for idim in range(self.ndim) if idim not in axis]
+            else:
+                chunk_shape = subset_state.to_mask(self, chunk_view).shape
+                full_shape = [chunk_shape[idim] for idim in range(self.ndim) if idim not in axis]
+                view_start = [chunk_view[idim].start for idim in range(self.ndim) if idim not in axis][0]
```

Comment:
```suggestion
```
This is never used anymore in this branch.

Reply:
Whoops, good catch.